packages/realm-server/prerender/manager-app.ts

```diff
@@ -0,0 +1,391 @@
+    if (url) {
+      url = normalizeURL(url);
+      let list = registry.realms.get(realm) || [];
+      let idx = list.indexOf(url);
+      if (idx >= 0) list.splice(idx, 1);
+      if (list.length === 0) registry.realms.delete(realm);
+      // free capacity marker
+      registry.servers.get(url)?.activeRealms.delete(realm);
+    } else {
+      log.warn('Cannot dispose realm: cannot infer prerender server url');
+    }
+    ctxt.status = 204;
+  });
+
+  // helper: choose server for realm
+  function chooseServerForRealm(realm: string): string | null {
+    let assigned = registry.realms.get(realm);
+    if (assigned && assigned.length > 0) {
+      // If we have fewer than multiplex servers assigned, try to add a new one and prefer returning it
+      if (assigned.length < multiplex) {
+        for (let [url, info] of registry.servers) {
+          if (
+            !assigned.includes(url) &&
+            info.activeRealms.size < info.capacity
+          ) {
+            assigned.push(url);
+            // prefer the newly added server to spread load
+            return url;
+          }
+        }
+      }
+      // Otherwise rotate among the assigned set
+      let next = assigned.shift()!;
+      assigned.push(next);
+      if (assigned.length > multiplex) {
+        assigned.splice(0, assigned.length - multiplex);
+      }
+      return next;
+    }
+    // pick server with available capacity
+    for (let [url, info] of registry.servers) {
+      if (info.activeRealms.size < info.capacity) {
+        // assign
+        let list = registry.realms.get(realm) || [];
+        if (!list.includes(url)) list.push(url);
+        if (list.length > multiplex) list = list.slice(-multiplex);
+        registry.realms.set(realm, list);
+        info.activeRealms.add(realm);
+        return url;
+      }
+    }
+    // pressure mode: pick server owning globally LRU realm
+    let lruRealm: string | undefined;
+    let lruTime = Infinity;
+    for (let [r, t] of registry.lastAccessByRealm) {
+      if (t < lruTime) {
+        lruTime = t;
+        lruRealm = r;
+      }
+    }
+    if (lruRealm) {
+      let arr = registry.realms.get(lruRealm);
+      let url = arr?.[0];
+      if (url) {
+        let list = registry.realms.get(realm) || [];
+        if (!list.includes(url)) list.push(url);
+        if (list.length > multiplex) list = list.slice(-multiplex);
+        registry.realms.set(realm, list);
+        // don't mark active until success
+        return url;
+      }
+    }
+    // fallback: any server
+    let any = [...registry.servers.keys()][0];
+    if (any) {
+      let list = registry.realms.get(realm) || [];
+      if (!list.includes(any)) list.push(any);
+      if (list.length > multiplex) list = list.slice(-multiplex);
+      registry.realms.set(realm, list);
+      return any;
+    }
+    return null;
+  }
+
+  // health sweep: remove unreachable servers and clean up realm mappings
+  async function sweepServers() {
```

Comment:
What actually calls this? I could only find it being used in a test.

Reply:
is it going to be a cron job?

Reply:
good catch--this was supposed to be called by a setInterval in the manager app. fixing....